src/cpu/core_dyn_x86/risc_x64.h

```diff
@@ -1303,7 +1304,8 @@ static void gen_dh_fpu_saveInit(void) {
 	cache_addb(0xC3); // RET
 	
 	dyn_mem_execute(cache_addr, cache_bytes);
-	dyn_cache_invalidate(cache_addr, cache_bytes);
+	const size_t cache_flush_bytes = static_cast<size_t>(cache.pos - oldpos);
+	dyn_cache_invalidate(cache_addr, cache_flush_bytes);
```

Comment:
:+1: for hitting all the cores, and consistent code.

The invalidation size appears to be refined / smaller too (instead of all of the `cache_size`, it's now only the chunk of changed cache)

Reply:
Right, I thought we could save wasted cache clearing in all cases, rather than brute forcing a dump of CACHE_MAXSIZE (12k) bytes, now it's down in the several hundreds for the average case!